src/python/pants/engine/target.py

```diff
@@ -638,21 +638,16 @@ def expect_single(self) -> Target:
 
 
 class CoarsenedTarget(EngineAwareParameter):
-    """A set of Targets which cyclicly reach one another, and are thus indivisible.
-
-    Instances of this class form a structure-shared DAG, and so a hashcode is pre-computed for the
-    recursive portion.
-    """
+    def __init__(self, members: Iterable[Target], dependencies: Iterable[CoarsenedTarget]) -> None:
+        """A set of Targets which cyclicly reach one another, and are thus indivisible.
 
-    # The members of the cycle.
-    members: FrozenOrderedSet[Target]
-    # The deduped direct (not transitive) dependencies of all Targets in the cycle. Dependencies
-    # between members of the cycle are excluded.
-    dependencies: FrozenOrderedSet[CoarsenedTarget]
-    # Pre-computed hashcode: see the class doc.
-    _hashcode: int
+        Instances of this class form a structure-shared DAG, and so a hashcode is pre-computed for the
+        recursive portion.
 
-    def __init__(self, members: Iterable[Target], dependencies: Iterable[CoarsenedTarget]) -> None:
+        :param members: The members of the cycle.
+        :param dependencies: The deduped direct (not transitive) dependencies of all Targets in
+            the cycle. Dependencies between members of the cycle are excluded.
+        """
         self.members = FrozenOrderedSet(members)
         self.dependencies = FrozenOrderedSet(dependencies)
```

Comment:
Hm. I just noticed that `FrozenOrderedSet` has a memoized hashcode. So... it might have been premature for me to have added `_hashcode` (although not too costly either).